### PR TITLE
Refactor/utils cleanup

### PR DESCRIPTION
**Pull request by Solpatium**

### CODE REVIEW

File: starknet_py/net/client.py [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
     JsonObject,
 )
 from starkware.starknet.services.api.gateway.gateway_client import GatewayClient
-from services.external_api.base_client import RetryConfig, BadRequest as BadRequestError
+from starkware.starkware_utils.error_handling import StarkErrorCode
 
 from starknet_py.constants import TxStatus, ACCEPTED_STATUSES
 from starknet_py.utils.sync import add_sync_methods
-from starknet_py.utils.types import net_address_from_net, InvokeFunction, Transaction
+from .models import InvokeFunction, Transaction, Deploy
```
Comment (Arcticae) [inline starknet_py/net/client.py:17]: i'd rather we used absolute imports everywhere

File: starknet_py/net/client.py [hunk 2]
```diff
@@ -213,3 +217,24 @@ async def add_transaction(
...
+    async def deploy(
+        self,
+        compiled_contract: Optional[str],
+        constructor_calldata: List[int],
+        salt: Optional[int] = None,
+    ) -> dict:
+        res = await self.add_transaction(
+            tx=Deploy(
+                contract_address_salt=ContractAddressSalt.get_random_value()
+                if salt is None
```
Comment (Arcticae) [inline starknet_py/net/client.py:230]: maybe `salt or ContractAddressSalt.get_random_value()` would be more readable

File: starknet_py/net/client.py [hunk 3]
```diff
@@ -213,3 +217,24 @@ async def add_transaction(
...
+                else salt,
+                contract_definition=ContractDefinition.loads(compiled_contract),
+                constructor_calldata=constructor_calldata,
+            )
+        )
+
+        if res["code"] != StarkErrorCode.TRANSACTION_RECEIVED.name:
+            raise Exception("Transaction not received")
```
Comment (Arcticae) [inline starknet_py/net/client.py:238]: Can we cover it with tests?

File: starknet_py/net/networks.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from typing import Union
+
+MAINNET = "mainnet"
+TESTNET = "testnet"
+
+try:
+    from typing import Literal  # pylint: disable=no-name-in-module
+
+    PredefinedNetwork = Literal["mainnet", "testnet"]
+except:  # pylint: disable=bare-except
```
Comment (Arcticae) [inline starknet_py/net/networks.py:10]: shouldn't we catch ImportError here only?

File: starknet_py/net/networks.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    PredefinedNetwork = str
+
+Network = Union[PredefinedNetwork, str]
+
+
+def net_address_from_net(net: Network) -> str:
```
Comment (Arcticae) [inline starknet_py/net/networks.py:16]: I think it should actually return Optional[str] since network can be a string

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: starknet_py/net/client.py (hunk 2 above)
Comment (Solpatium) [reply starknet_py/net/client.py:230]: salt can be `0`

**Pull request merged**